Assets/Scripts/Communication/Server.cs

```diff
@@ -0,0 +1,128 @@
+﻿using UnityEngine;
+using System;
+using System.Text;
+using System.Collections.Generic;
+using Common.Cryptography;
+using System.Security.Cryptography;
+using System.Text.RegularExpressions;
+
+// This class contains all of the Server call functions
+public static class Server {
+
+	//private const float SERVER_TIMEOUT = 5f;			// Seconds before server call times out
+
+	//private static bool _timedOut;
+
+	public static void Connect() {
+		Communication.Connect(new Uri("ws://localhost:8000"));
+	}
+
+	/*public static IEnumerator Timeout(int timeout) {
+		Server._timedOut = false;
+		Debug.Log("Trying to connect to server...");
+		int seconds = 0;
+		while(seconds < timeout) {
+			seconds++;
+			yield return new WaitForSeconds(1f);
+			Debug.Log("Waited " + seconds + " of " + Server.SERVER_TIMEOUT + " seconds...");
+			if(seconds == timeout) {
+				Server._timedOut = true;
+				break;
+			}
+		}
+	}*/
+
+	// Used to login to server with username and password
+	public static bool Login(string username, string pw) {
+		// Create the request, set data to pass, and send it
+		var request = new Dictionary<string, object>();
+		request["Command"] 	= "LGN";
+		request["username"]	= username;
+		request["pw"]		= pw;
+		Communication.SendString(Json.ToString(request));
+		// Wait for the response, then parse
+		string strResponse = null;
+		while(strResponse == null) {
+			strResponse = Communication.RecvString();
+		}
+		var response = Json.ToDict(strResponse);
+		// If login successful, encrypt and store session token
+		bool success = (bool)response["Success"];
+		if(success) {
+			string _loginToken = response["Token"].ToString();
+			string _encryptedToken = AES.Encrypt(_loginToken, GenerateAESKey());
+			PlayerPrefs.SetString("session", _encryptedToken);
+			Debug.Log("user '" + username + "' logged in with token: " + _loginToken);
+		}
+		return success;
+	}
+
+	// Used to login to server with cached session token
+	public static bool RetryLogin() {
+		// Create the request, decrypt session token, and send it
+		var request = new Dictionary<string, object>();
+		string _encryptedToken = PlayerPrefs.GetString("session");
+		string _loginToken = AES.Decrypt(_encryptedToken, GenerateAESKey());
```

Comment:
Probably need a check here to see if we are able to get the session token successfully from playerprefs.  If it doesn't exist there, we should fall back to the normal login with username and password.  That can come later, just wanted to point that out.